src/highlight.js

```diff
@@ -4,692 +4,248 @@ https://highlightjs.org/
 */
 
 // @ts-ignore
-import deepFreeze from 'deep-freeze-es6';
-import Response from './lib/response.js';
-import TokenTreeEmitter from './lib/token_tree.js';
-import * as regex from './lib/regex.js';
-import * as utils from './lib/utils.js';
-import * as MODES from './lib/modes.js';
-import { compileLanguage } from './lib/mode_compiler.js';
-import * as packageJSON from '../package.json';
-import * as logger from "./lib/logger.js";
-import HTMLInjectionError from "./lib/html_injection_error.js";
-
-
-/**
-@typedef {import('highlight.js').Mode} Mode
-@typedef {import('highlight.js').CompiledMode} CompiledMode
-@typedef {import('highlight.js').CompiledScope} CompiledScope
-@typedef {import('highlight.js').Language} Language
-@typedef {import('highlight.js').HLJSApi} HLJSApi
-@typedef {import('highlight.js').HLJSPlugin} HLJSPlugin
-@typedef {import('highlight.js').PluginEvent} PluginEvent
-@typedef {import('highlight.js').HLJSOptions} HLJSOptions
-@typedef {import('highlight.js').LanguageFn} LanguageFn
-@typedef {import('highlight.js').HighlightedHTMLElement} HighlightedHTMLElement
-@typedef {import('highlight.js').BeforeHighlightContext} BeforeHighlightContext
-@typedef {import('highlight.js/private').MatchType} MatchType
-@typedef {import('highlight.js/private').KeywordData} KeywordData
-@typedef {import('highlight.js/private').EnhancedMatch} EnhancedMatch
-@typedef {import('highlight.js/private').AnnotatedError} AnnotatedError
-@typedef {import('highlight.js').AutoHighlightResult} AutoHighlightResult
-@typedef {import('highlight.js').HighlightOptions} HighlightOptions
-@typedef {import('highlight.js').HighlightResult} HighlightResult
-*/
-
-
-const escape = utils.escapeHTML;
-const inherit = utils.inherit;
-const NO_MATCH = Symbol("nomatch");
-const MAX_KEYWORD_HITS = 7;
-
-/**
- * @param {any} hljs - object that is extended (legacy)
- * @returns {HLJSApi}
- */
+import deepFreeze from 'deep-freeze-es6'; // Prevents objects from being modified
+import Response from './lib/response.js'; // Handles responses
+import TokenTreeEmitter from './lib/token_tree.js'; // Manages token trees for highlighting
+import * as regex from './lib/regex.js'; // Contains regex utilities
+import * as utils from './lib/utils.js'; // Contains utility functions
+import * as MODES from './lib/modes.js'; // Different modes for syntax highlighting
+import { compileLanguage } from './lib/mode_compiler.js'; // Compiles language definitions
+import * as packageJSON from '../package.json'; // Package information
+import * as logger from "./lib/logger.js"; // Logging utility
+import HTMLInjectionError from "./lib/html_injection_error.js"; // Error handling for HTML injection
+
+const escape = utils.escapeHTML; // Function to escape HTML
+const inherit = utils.inherit; // Function to inherit properties
+const NO_MATCH = Symbol("nomatch"); // Symbol for no match found
```

Comment:
There MAY be some room for improving the readability by improving the naming, for example here `NO_MATCH_FOUND` would likely be a clearer name - but again the name says it all - we don't need a comment that just effectively repeats the name.